inveniordm_py/records/resources.py

```diff
@@ -0,0 +1,295 @@
+        """Update a draft."""
+        return self._put(DraftMetadata, data=data or self.data)
+
+    @property
+    def files(self):
+        """Draft files."""
+        return DraftFilesList(self._client, **self.endpoint_args)
+
+    def import_files(self):
+        """Run import files action."""
+        # TODO
+        self._post(
+            None,
+        )
+        resp = self.session.post(
+            self.url("/actions/files-import"), headers=self.headers({})
+        )
+        self.raise_on_error(resp)
+        self.data = self.item_class(resp)
+        return self
+
+    def publish(self):
+        """Publish draft."""
+        return self._post(
+            RecordMetadata,
+            url_suffix="/actions/publish",
+            resource=Record(self._client, **self.endpoint_args),
+        )
+
+
+class RecordList(Resource):
+    """Implements a RecordList as a Resource.
+
+    This is the resource that is used to interact with the /api/records endpoint.
+    """ ""
+
+    endpoint = "/records"
+
+    def __call__(self, id_):
+        """Instantiate a record item resource."""
+        return Record(self._client, id_=id_)
+
+    @property
+    def draft(self):
+        """Creates and returns a record draft API object."""
+        return Draft(self._client, **self.endpoint_args)
+
+    def create(self, data=None):
+        """Create new draft."""
+        return self._post(DraftMetadata, data=data, resource=self.draft)
+
+    def search(self, q="", page=1, size=10, sort="newest", allversions=False):
+        """Search for records."""
+        params = dict(q=q, page=page, size=size, sort=sort)
+        if allversions:
+            params["allversions"] = "1"
+        return self._search(
+            params,
+            RecordListMetadata,
+            self._make_factory(Record),
+            self._partial(self.search, params, page=params["page"] - 1),
+            self._partial(self.search, params, page=params["page"] + 1),
+        )
+
+
+class RecordFilesList(Resource):
+    """Implements a RecordFilesList as a Resource.
+
+    This is the resource that is used to interact with the /api/records/{id_}/files endpoint.
+    """
+
+    endpoint = "/records/{id_}/files"
+
+    def get(self):
+        """Get a files lsit"""
+        return self._get(FilesListMetadata)
+
+    def __call__(self, key):
+        """Instantiate a record item resource."""
+        return RecordFile(self._client, filename=key, **self._endpoint_args)
+
+
+class RecordFile(Resource):
+    """Implements a RecordFile as a Resource.
+
+    This is the resource that is used to interact with the /api/records/{id_}/files/{filename} endpoint.
+    """
+
+    endpoint = "/records/{id_}/files/{filename}"
+
+    def get(self):
+        """Get file metadata."""
+        return self._get(FileMetadata)
+
+    def download(self):
+        """Download a file"""
+        return self._get_raw(IncomingStream, url_suffix="/content", params={"stream": True})
```

Comment:
I am not sure about this implementation. 

The tool is designed around the concept of a "Resource", which is a 1 to 1 match with RDM's resources (i.e. Records, Drafts, etc). However, to download a file we don't get any RDM-equivalent "resource". That being said, using the resource's HTTP methods wrappers (`GET / POST`) doesn't cater for this use case.